tests/components/gogogate2/test_cover.py

```diff
@@ -470,3 +416,73 @@ async def test_availability(ismartgateapi_mock, hass: HomeAssistant) -> None:
     async_fire_time_changed(hass, utcnow() + timedelta(hours=2))
     await hass.async_block_till_done()
     assert hass.states.get("cover.door1").state == STATE_CLOSED
+
+
+@patch("homeassistant.components.gogogate2.common.ISmartGateApi")
+async def test_device_info_ismartgate(ismartgateapi_mock, hass: HomeAssistant) -> None:
+    """Test device info."""
+    device_registry = mock_device_registry(hass)
+
+    closed_door_response = _mocked_ismartgate_closed_door_response()
+
+    api = MagicMock(ISmartGateApi)
+    api.info.return_value = closed_door_response
+    ismartgateapi_mock.return_value = api
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        source=SOURCE_USER,
+        title="mycontroller",
+        unique_id="xyz",
+        data={
+            CONF_DEVICE: DEVICE_TYPE_ISMARTGATE,
+            CONF_IP_ADDRESS: "127.0.0.1",
+            CONF_USERNAME: "admin",
+            CONF_PASSWORD: "password",
+        },
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    device = device_registry.async_get_device({(DOMAIN, "xyz")}, set())
+    assert device
+    assert device.manufacturer == MANUFACTURER
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)